libindy_vdr/src/common/handle.rs

```diff
@@ -0,0 +1,47 @@
+pub trait ResourceHandle: Copy + Ord + From<i64> {
+    fn invalid() -> Self {
+        Self::from(0)
+    }
+
+    fn next() -> Self;
+}
+
+/// Derive a new handle type having an atomically increasing sequence number
+macro_rules! new_sequence_handle (($newtype:ident, $counter:ident) => (
```

Comment:
I am not too sure how, but can we expose the handle type so we do not have to add it manually in the header? so the `$newtype` would expose it to the header as `typedef int64_t $newtype;`